proposals/v3/api.md

```diff
@@ -0,0 +1,815 @@
+          "type": "periodic"
+        }
+      ]
+    }
+  }
+]
+```
+
+**Fields**
+
+| Field | Description |
+| ----- | ----------- |
+| *id* | An id hash for identifying the plugin, generated from plugin metadata. |
+| *tag* | The plugin tag. This is a normalized string made up of its name and maintainer. |
+| *name* | The name of plugin. |
+| *maintainer* | The maintainer of the plugin. |
+| *description* | A short description of the plugin. |
+| *vcs* | A link to the version control repo for the plugin. |
+| *version* | An object that contains version information about the plugin. |
+| *version.plugin_version* | The plugin version. |
+| *version.sdk_version* | The version of the Synse SDK that the plugin is using. |
+| *version.build_date* | The date that the plugin was built. |
+| *version.git_commit* | The git commit at which the plugin was built. |
+| *version.git_tag* | The git tag at which the plugin was built. |
+| *version.arch* | The architecture that the plugin is running on. |
+| *version.os* | The OS that the plugin is running on. |
+| *network* | An object that describes the network configurations for the plugin. |
+| *network.protocol* | The protocol that is used to communicate with the plugin (unix, tcp). |
+| *network.address* | The address of the plugin for the protocol used. |
+| *health* | An object that describes the overall health of the plugin. |
+| *health.timestamp* | The time at which the health status applies. |
+| *health.status* | The health status of the plugin (ok, degraded, failing, error, unknown) |
+| *health.message* | A message describing the error, if in an error state. |
+| *health.checks* | A collection of health check snapshots for the plugin. |
+
+
+There may be 0..N health checks for a Plugin, depending on how it is configured.
+The health check elements here make up a snapshot of the plugin's health at a given time.
+
+| Field | Description |
+| ----- | ----------- |
+| *name* | The name of the health check. |
+| *status* | The status of the health check (ok, failing) |
+| *message* | A message describing the failure, if in a failing state. |
+| *timestamp* | The timestamp for which the status applies. |
+| *type* | The type of health check (e.g. periodic) |
+
+#### Error (500)
+See: [Errors](#errors)
+
+
+
+## Scan
+```
+GET http://HOST:5000/synse/v3/scan
+```
+
+List the devices that Synse knows about and can read from/write to via
+the configured plugins.
+
+This endpoint provides an aggregated view of the devices made known to Synse
+Server by each of the registered plugins. This endpoint provides a high-level
+view of what exists in the system. Scan info can be filtered to show only those
+devices which match a set of provided tags.
+
+The scan results are sorted by device id.
+
+### Query Parameters
+
+| Key | Description |
+| --- | ----------- |
+| namespace | The default namespace to use for the specified labels. (default: `default`) |
+| ns | Same as `namespace`, above. |
```

Comment:
Yup. I was initially thinking that having the alias would be convenient, since typing `ns` is easier than `namespace`, but it would probably just complicate backend processing more than necessary, so we should just choose one and stick with it. I'd say we use `ns`, but I don't have a strong preference either way. Thoughts?

Reply:
Even though having both `namespace` and `ns` adds a bit of complexity, it's pretty intuitive and friendly to use. I think it's fine. But if I have to choose one, I would go with `ns` too.